ql/src/java/org/apache/hadoop/hive/ql/queryhistory/repository/IcebergRepository.java

```diff
@@ -176,4 +187,9 @@ private void prepareConfForWrite() {
     SessionStateUtil.addCommitInfo(SessionState.getSessionConf(), tableDesc.getTableName(), jobId, 1,
         Maps.fromProperties(tableDesc.getProperties()));
   }
+
+  private void expireSnapshots() {
+    storageHandler.executeOperation(table, new AlterTableExecuteSpec<>(EXPIRE_SNAPSHOT,
+        new AlterTableExecuteSpec.ExpireSnapshotsSpec(10)));
```

Comment:
Do you want to hardcode ``10`` here or just pass ``null`` & let the TBLPROPERTIES handle it, this way it can be configurable also by ALTER command.

I don't think this would work the way you expect, there is some default time thing as well
```
  public static final String MAX_SNAPSHOT_AGE_MS = "history.expire.max-snapshot-age-ms";
  public static final long MAX_SNAPSHOT_AGE_MS_DEFAULT = 432000000L;
```

Reply:
thanks @ayushtkn for taking a look!
good point, I'm not familiar with EXPIRE SNAPSHOT functionality detail, so this is more like a DRAFT, let me consider

Reply:
this seemed to work:
https://github.com/apache/hive/pull/5666/commits/4c9480ff53ab69e8a795a620d0ba8d5f9c635dec

added testing details in PR description

Reply:
@ayushtkn : I just added a unit test for snapshot expiry testing